File: SmartX/app/src/main/java/com/smartx/cookies/smartx/Device.java
package com.smartx.cookies.smartx;

/**
 * Created by zamzamy on 3/18/15.
 */
public class Device {
    int userID;
    String type;
    String name;
    String brand;
    int roomID;

    public Device(){

    }

    public Device(String type, String name, String brand, int roomID) {
        this.type = type;
        this.name = name;
        this.brand = brand;
        this.roomID = roomID;
    }
}

File: api-rails/app/models/device.rb
class Device < ActiveRecord::Base
	before_save :default_values
  def default_values
    self.favorite ||= "false"
  end
	self.primary_key = 'device_id'
	belongs_to :rooms
	has_one :clicker
	has_many :notes,dependent: :destroy
	validates :name, :uniqueness => {:scope => [:user_id, :room_id]}
	def to_param
    	device_id.parameterize
  	end
end
